File: WordPig/app/src/main/java/com/pigdogbay/wordpig/model/WordChecker.java
package com.pigdogbay.wordpig.model;

import java.util.List;

/**
 * Created by Mark on 02/04/2015.
 */
public class WordChecker
{
    List<String> wordList;

    public WordChecker(List<String> wordList)
    {
        this.wordList = wordList;
    }

    public boolean isWord(String candidate)
    {
//        int len = candidate.length();
        for (String word : wordList)
        {
            if (word.equals(candidate))
            {
                return true;
            }
        }
        return false;
    }
}

File: WordPig/app/src/main/java/com/pigdogbay/wordpig/model/WordChecker.kt
package com.pigdogbay.wordpig.model

/**
 * Created by Mark on 02/04/2015.
 */
class WordChecker(private var wordList: List<String>) {
    fun isWord(candidate: String): Boolean {
        for (word in wordList) {
            if (word == candidate) {
                return true
            }
        }
        return false
    }
}
